### PR TITLE
Allow generating widgets without name.

### PR DESCRIPTION
**Pull request by ADmad**
If widget name is empty preclude it from secure fields list.

I am making a custom widget which consists of a basic text input and select input. Only the basic input needs to be actually POSTed and the select is only for display purpose required by jquery plugin. Without this change i am unable to keep "name" attribute of the select tag empty to prevent it from being POSTed.

I also ensured for all widgets which can have empty name that `secureFields()` should return empty array.

TL;DR;  Inputs without names are valid and should not cause request to be blackholed on form post.

### CODE REVIEW

**Comment by jadb** [comment]
nice catch! :clap:

**Comment by markstory** [comment]
Alternatively, secureFields() could filter out ''.

**Comment by ADmad** [comment]
@markstory That could work too. In that case only the exception would need to be removed from `SelectBoxWidget`.

**Comment by ADmad** [comment]
Umm which `secureFields()` are you referring to? FormHelper does not have a `secureFields()` and i have already updated `secureFields()` of widgets.

**Comment by ADmad** [comment]
Opened an alternate PR which might be more preferable than this.
